pest-ion/src/bin/pest2ion/main.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+use anyhow::Result;
+use clap::{crate_authors, crate_description, crate_version, App, Arg, ArgGroup};
+use ion_rs::value::writer::{ElementWriter, Format, TextKind};
+use pest_ion::{from_read, TryPestToElement};
+use std::fs::File;
+use std::io::{stdin, stdout, Write};
+use std::path::Path;
+
+fn main() -> Result<()> {
+    let matches = App::new("Pest to Ion Converter")
+        .version(crate_version!())
+        .author(crate_authors!())
+        .about(crate_description!())
+        .arg(
+            Arg::with_name("FILE")
+                .help("The input file to parse (defaults to STDIN)")
+                .index(1),
+        )
+        .arg(
+            Arg::with_name("OUTPUT_FILE")
+                .long("output")
+                .short("o")
+                .takes_value(true)
+                .help("Writes output to the given file (defaults to STDOUT)"),
+        )
+        .arg(
+            Arg::with_name("text")
+                .long("text")
+                .short("t")
+                .help("Generate Ion text (default)"),
+        )
+        .arg(
+            Arg::with_name("binary")
+                .long("binary")
+                .short("b")
+                .help("Generate Ion binary"),
+        )
+        .arg(
+            Arg::with_name("pretty")
+                .long("pretty")
+                .short("p")
+                .help("Generate Ion pretty printed text"),
+        )
+        .group(ArgGroup::with_name("format").args(&["text", "binary", "pretty"]))
```

Comment:
I'm unsure how this works. Do I need to do `--format=p`? For reference, this is [how the `ion-cli` does it](
https://github.com/amzn/ion-cli/blob/4ff67c313b9f2c00c73a4f4cc9a21d0ef009d263/src/bin/ion/commands/dump.rs#L48-L54):

```rust
            Arg::with_name("format")
                .long("format")
                .short("f")
                .takes_value(true)
                .default_value("pretty")
                .possible_values(&["binary", "text", "pretty"])
                .help("Output format"),
```

Reply:
The reference is [here](https://docs.rs/clap/2.33.3/clap/struct.ArgGroup.html).  Basically I have a mutual exclusion to `-p`/`-b`/`-t` flags.  If you think we should use an option flag to be more consistent with `ion-cli` (which is marginally verbose), I am not opposed.

Reply:
`ion-cli` takes the slightly more verbose route (`-f binary`) largely to preserve flag name options for other use cases. Virtually every command needs to support different Ion formats. Some may have really good specialized uses for `-b`, `-p`, and `-t`. `pest2ion` is much more limited in scope, so consuming the shortcut namespace isn't as much of an issue.

Reply:
Yeah, to be honest, I only exposed these flags because of #43, there is an underlying bug in `ion-c-sys`, so instead of hard coding to binary, I decided to add these.

My question to you, should we make this consistent?  Not opposed, though not really planning on adding this as an `ion-cli` sub-command.

Reply:
> My question to you, should we make this consistent? Not opposed, though not really planning on adding this as an ion-cli sub-command.

I think you can leave it as-is. If this becomes widely used enough to justify adding it to `ion-cli` we can always tweak the interface in that version for consistency.